tools/find-non-ascii-chars-in-code/find-non-ascii-in-code.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func isNonASCII(s string) bool {
+	for _, r := range s {
+		if r > 127 {
+			return true
+		}
+	}
+	return false
+}
+
+func checkIdentifiersInFile(path string) (bool, error) {
+	fset := token.NewFileSet()
+	node, err := parser.ParseFile(fset, path, nil, parser.SkipObjectResolution)
+	if err != nil {
+		return false, err
+	}
+
+	hasViolation := false
+
+	// Walk the AST
+	ast.Inspect(node, func(n ast.Node) bool {
+		switch v := n.(type) {
+		case *ast.Ident: // inspect only the identifiers, ignoring string literals and comments
+			if isNonASCII(v.Name) {
```

Comment:
I think it is possible to make it a bit simpler:
```suggestion
		v, ok := n.(*ast.Ident) 
		if ok && isNonASCII(v.Name) {
```